src/directives/profile.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @fileoverview Provides a directive used to insert an elevation profile chart
+ * in the DOM.
+ *
+ * Example:
+ *
+ * <div ngeo-profile="ctrl.profileData"
+ *      ngeo-profile-onhover="ctrl.onHover(point)"
+ *      ngeo-profile-onout="ctrl.onProfileOut()"></div>
+ */
+goog.provide('ngeo.profileDirective');
+
+goog.require('ngeo');
+goog.require('ngeo.profile');
+
+
+/**
+ * @return {angular.Directive} Directive Definition Object.
+ */
+ngeo.profileDirective = function() {
+  return {
+    restrict: 'A',
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Attributes.
+         */
+        function(scope, element, attrs) {
+
+          var selection = d3.select(element[0]);
+          var profile = ngeo.profile();
+
+          scope.$watch(attrs['ngeoProfile'], function(newVal, oldVal) {
+            if (goog.isDef(newVal)) {
+              selection.datum(newVal['profile']).call(profile);
```

Comment:
Would it be possible to get directly the profile data with `selection.datum(newVal).call(profile);`
and write the html like `<div class="profile" ngeo-profile="ctrl.profileData.profile"...`?